orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactUtils.java

```diff
@@ -227,10 +230,45 @@ public List<Artifact> getArtifactsForPipelineIdWithoutStageRef(
         .orElse(Collections.emptyList());
   }
 
+  private List<String> getExpectedArtifactIdsFromMap(Map<String, Object> trigger) {
+    List<String> expectedArtifactIds = (List<String>) trigger.get("expectedArtifactIds");
+    return (expectedArtifactIds != null) ? expectedArtifactIds : emptyList();
+  }
+
   public void resolveArtifacts(Map pipeline) {
     Map<String, Object> trigger = (Map<String, Object>) pipeline.get("trigger");
-    List<?> expectedArtifactIds =
-        (List<?>) trigger.getOrDefault("expectedArtifactIds", emptyList());
+    List<?> triggers = Optional.ofNullable((List<?>) pipeline.get("triggers")).orElse(emptyList());
+    Set<String> expectedArtifactIdsListConcat =
+        new HashSet<>(getExpectedArtifactIdsFromMap(trigger));
+
+    // Due to 8df68b79cf1 getBoundArtifactForStage now does resolution which can
+    // potentially return an empty list or null back, which will throw an
+    // exception. Before this commit, when getBoundArtifactForStage was called,
+    // the method would just retrieve the bound artifact from the stage context,
+    // and return it as an ExpectedArtifact type instead of a map. "Resolution"
```

Comment:
Not sure I understand the `instead of a map` part.    Looking at https://github.com/spinnaker/orca/commit/8df68b79cf103e954f305c342fbc33733f26b8f2#diff-b25c50ecaceee2f8b0ae1ddc77fc92d48d16fb7ff9d15d27612685f1985b2a1eL136-R144, didn't both the "before" and "after" code return an Artifact?

This means I don't understand how getBoundArtifactForStage can return an empty list since the return type is `Artifact`.

I'm also a little unclear why there's a comment here, in `resolveArtifacts` about the behavior of `getBoundArtifactForStage`.  I could use a bit more connecting of the dots.